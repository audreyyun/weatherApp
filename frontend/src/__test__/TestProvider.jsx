import React from 'react'
import { Provider } from 'react-redux';
import configureStore from '../store/store';

const store = configureStore();

const TestProvider = ({ children }) => (
    <Provider store={store}>
        {children}
    </Provider>
);

export default TestProvider;