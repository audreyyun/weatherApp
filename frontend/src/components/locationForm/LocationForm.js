import React, { useState } from 'react';
import useLocation from '../../hooks/useLocation';

const geolocationOptions = {
    timeout: 1000 * 60 * 1
};

const LocationForm = ({ fetchWeather }) => {
    console.log("I am in input", fetchWeather)
    const [city, setCity] = useState('');
    const {location, error } = useLocation(geolocationOptions)


    const handleCityChange = e => {
        console.log(e)
        setCity(e.target.value)
    };

    if (location && !city) { 
        setCity(location);
        fetchWeather(location);
    };

    const handleSubmit = () => { 
        fetchWeather(city)
    };

    return (
        <div>
            <div className="location-form">
                <div className="location-input">
                    <input className='zipcode-search-input input' onChange={handleCityChange} placeholder=' Enter City Name'/>
                    <button className="input location-input-btn" onClick={handleSubmit}>See Weather</button>
                </div>
            </div>
        </div>
    )
}
export default LocationForm;