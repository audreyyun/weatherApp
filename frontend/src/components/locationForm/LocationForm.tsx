import React, { useCallback, useEffect, useState } from 'react';
import classNames from 'classnames';
import useLocation from '../../hooks/useLocation';
import styles from './LocationForm.module.scss'

const geolocationOptions = {
    timeout: 1000 * 60 * 1
};

type Props = {
    fetchWeather: Function
}

const LocationForm = ({ fetchWeather } : Props) => {
    const [city, setCity] = useState<String | undefined>('');
    //decide where/when to render error for geolocation 
    const { location, error } = useLocation(geolocationOptions);


    const handleCityChange = (e: any) => {
        setCity(e.target.value)
    };

    const handleFetchWeather = useCallback( async (location) => { 
                fetchWeather(location) 
    },[]);

    useEffect(() => {
            setCity(location);
            handleFetchWeather(location)
    },[location]);
        
    const handleSubmit = () => { 
        fetchWeather(city)
    };

    return (
        <div>
            <div className={styles.locationForm}>
                <div className={styles.locationInput}>
                    <input className={classNames(styles.zipcodeSearchInput, styles.input)} onChange={handleCityChange} placeholder=' Enter City Name'/>
                    <button className={classNames(styles.locationInputBtn, styles.input)} onClick={handleSubmit}>See Weather</button>
                </div>
            </div>
        </div>
    )
};

export default LocationForm;