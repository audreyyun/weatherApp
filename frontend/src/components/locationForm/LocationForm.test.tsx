import '@testing-library/jest-dom/extend-expect';
import { render, fireEvent } from '@testing-library/react'
import React from 'react'
import LocationForm from './LocationForm'
import TestProvider from '../../__test__/TestProvider';


const LocationFormTest = () => (
    <TestProvider>
        <LocationForm />
    </TestProvider>
);


describe('LocationForm', () => {
    it('should render without crashing', () => {
        const component = render(<LocationFormTest />)
        expect(component.asFragment()).toMatchSnapshot()
    })
})


it('accepts an input value', () => {
    const component = render(<LocationForm />)
    
    const searchInput = component.getByPlaceholderText('Enter City Name') as HTMLInputElement
    fireEvent.input(searchInput, { target: { value: 'Los Angeles' } })
    expect(searchInput.value).toBe('Los Angeles')
})