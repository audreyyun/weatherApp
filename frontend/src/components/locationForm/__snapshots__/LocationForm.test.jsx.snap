// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocationForm should render without crashing 1`] = `
<DocumentFragment>
  <div>
    <div
      class="locationForm"
    >
      <div
        class="locationInput"
      >
        <input
          class="zipcodeSearchInput input"
          placeholder=" Enter City Name"
        />
        <button
          class="locationInputBtn input"
        >
          See Weather
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
